src/SignupForm.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+
+class SignupForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      firstName: '',
+      lastName: '',
+      password: '',
+      agreeWithTerms: false,
+    };
+
+    this.handleEmailChange = this.handleEmailChange.bind(this);
```

Comment:
You can use array functions instead of binding them in constructor: these are bound instantly to component's context

Reply:
Ok, I will change that.